packages/taquito-wallet-connect/src/types.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  RPCDelegateOperation,
+  RPCRegisterGlobalConstantOperation,
+  RPCOriginationOperation,
+  RPCRevealOperation,
+  RPCTransferOperation,
+  RPCTransferTicketOperation,
+  RPCIncreasePaidStorageOperation,
+  RPCBallotOperation,
+  RPCUpdateConsensusKeyOperation,
+  RPCDrainDelegateOperation,
+  RPCProposalsOperation,
+  RPCSmartRollupAddMessagesOperation,
+  RPCSmartRollupOriginateOperation,
+  RPCSmartRollupOutboxMessageOperation,
+  RPCFailingNoopOperation,
+} from '@taquito/taquito';
+
+export enum NetworkType {
+  MAINNET = 'mainnet',
+  GHOSTNET = 'ghostnet',
+  WEEKLYNET = 'weeklynet',
+  OXFORDNET = 'oxfordnet',
+  PARISNET = 'parisnet',
+  // QUEBECNET = 'quebecnet',
+}
+
+export interface PermissionScopeParam {
+  networks: NetworkType[];
+  methods: PermissionScopeMethods[];
+  events?: PermissionScopeEvents[];
+}
+export enum PermissionScopeMethods {
+  TEZOS_GET_ACCOUNTS = 'tezos_getAccounts',
+  TEZOS_SEND = 'tezos_send',
+  TEZOS_SIGN = 'tezos_sign',
+}
+
+export enum PermissionScopeEvents {
+  CHAIN_CHANGED = 'chainChanged',
+  ACCOUNTS_CHANGED = 'accountsChanged',
+}
+
+export enum SigningType {
+  RAW = 'raw',
+  OPERATION = 'operation',
+  MICHELINE = 'micheline',
+}
+
+export interface TezosAccount {
+  algo: string;
+  address: string;
+  pubkey: string;
+}
+
+type WalletDefinedFields = 'source' | 'gas_limit' | 'storage_limit' | 'fee';
+interface WalletOptionalFields {
+  gas_limit?: string;
+  storage_limit?: string;
+  fee?: string;
+}
+
+export interface TransferParams
```

Comment:
Wondering if it would make sense to move those types so we can use them in beacon package too

Reply:
@roxaneletourneau Thank you, that'll be a great idea but it seems @airgap/beacon-types has their own type `PartialTezosOperation` and our walletAPI has type `WalletParamsWithKind`. 
This also made me realize that even in walletconnect package supports all contractAPI operations but walletAPI doesn't support them that their use will be limited. 
I'll merge this PR for now and start a new pr to support all contractAPI ops in walletAPI and consolidate the types